mp4parse_capi/src/lib.rs

```diff
@@ -741,13 +741,12 @@ fn create_sample_table(track: &Track) -> Option<Vec<mp4parser_indice>> {
                 }
             true
         });
-    if !ret {
+    if sample_table.is_empty() {
```

Comment:
It's my bad without explaining it early. My original idea is returning false if any of ```end_offset``` was zero. So when anything wrong in ```stsz.sample_sizes``` or ```stsc.samples```, we can identify it quickly.

I'm not quite sure your second ```.map()```, do you mean codes like this?
```
    stsc.samples.iter()
        .map(|sample_chunk| {
                // Convert the compact table to chunk range and samples per trunk.
                start_chunk = end_chunk;
                end_chunk += sample_chunk.first_chunk;
                ((start_chunk .. end_chunk), sample_chunk.samples_per_chunk)
            })
        .map(|chunks| {
                // Chunks in the chunk range have the same sample number.
                let chunks_range = chunks.0;
                let samples_per_chunk = chunks.1;
                for chunk_id in chunks_range {
                    let mut cur_position: u64 = stco.offsets[chunk_id as usize];
                    // Create sample according to sample number of each trunk and calculate
                    // sample offset.
                    for _ in 0 .. samples_per_chunk {
                        let start_offset = cur_position;
                        let end_offset = match (stsz.sample_size, sample_size_iter.next()) {
                            (_, Some(t)) => start_offset + *t as u64,
                            (t, _) if t > 0 => start_offset + t as u64,
                            _ => 0,
                        };
                        if end_offset > 0 {
                            cur_position = end_offset;

                            sample_table.push(
                                mp4parse_indice {
                                    start_offset: start_offset,
                                    end_offset: end_offset,
                                    start_composition: 0,
                                    end_composition: 0,
                                    start_decode: 0,
                                    sync: !has_sync_table,
                                }
                            );
                        }
                    }
                }
        });

    if sample_table.is_empty() {
        return None;
    }
```